### PR TITLE
setup: pin moto to under 2.0

### PR DESCRIPTION
**Pull request by isidentical**
I didn't had chance to debug it fully though moto>2.0 is incompatible with the tests as is (2 version check tests are failing) intake/filesystem_spec/pull/557

### CODE REVIEW

**Comment by martindurant** [comment]
( cc @mariusvniekerk , in case you know how versioning is different in moto 2.0 - no need for any action here)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -33,7 +33,8 @@ jobs:
       - name: Install dependencies
         shell: bash -l {0}
         run: |
-          conda install -c conda-forge pip botocore aiobotocore moto pytest flake8 black -y
+          conda install -c conda-forge pip botocore aiobotocore flake8 black -y
```
Comment (martindurant) [inline .github/workflows/ci.yml:36]: why not include the moto version pin in the conda install line? pip could in theory step on conda packages, so we should keep conda-based tests and pip-based tests separate.

Comment (isidentical) [reply .github/workflows/ci.yml:36]: I just thought maintaining the same list in 2 different places was redundant, though can revert.

Comment (martindurant) [reply .github/workflows/ci.yml:36]: True, but I don't see a way around it when I trust conda more than pip

Comment (isidentical) [reply .github/workflows/ci.yml]: alright dropped

**Comment by martindurant** [comment]
Thanks!

**Pull request merged**